src/init.cpp

```diff
@@ -1808,6 +1808,22 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
                     strLoadError = _("Corrupted block database detected");
                     break;
                 }
+
+                {
+                    LOCK(cs_main);
+
+                    SaplingMerkleTree sapling_tree;
+                    assert(pcoinsdbview->GetSaplingAnchorAt(pcoinsdbview->GetBestAnchor(SAPLING), sapling_tree));
+
+                    if (!SubtreeStateConsistent(SAPLING, sapling_tree, pcoinsdbview)) {
+                        LogPrintf("NOTE: the database does not contain some subtree data for Sapling or Orchard; most users will not need this, but if you do, you will have to use -reindex\n");
```

Comment:
```suggestion
                        LogPrintf("NOTE: the database does not contain complete subtree data for Sapling or Orchard; most users will not need this, but if you do, you will have to use -reindex\n");
```

Reply:
Say explicitly that this is needed for zcashd to back a lightwalletd instance.

Reply:
+1. If I read "most users" does that help me answer "is that me or not?"

Spell out the use cases impacted so users can identify if that's them.